src/deploy/functions/params.ts

```diff
@@ -0,0 +1,339 @@
+import { logger } from "../../logger";
+import { FirebaseError } from "../../error";
+import { promptOnce } from "../../prompt";
+import * as build from "./build";
+import { assertExhaustive } from "../../functional";
+
+type CEL = build.Expression<string> | build.Expression<number> | build.Expression<boolean>;
+
+function isCEL(expr: string | number | boolean): expr is CEL {
```

Comment:
Again, can the type of what's passed in be `unknown`? It's also odd a number or boolean could be passed in, even though they'd never pass the check

```suggestion
function isCEL(expr: unknown): expr is CEL {
```